### ISSUE TITLE
Adjust console keys on international keyboard

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#937]
Author: still (still)
Date: 2016-09-14
Redmine Issue: 937, https://dev.etlegacy.com/issues/937

---

Give back the consol openening in the standard way (with button under the ESC button)
Now not working, only work the shift+esc combo

---

Related issue(s): #274

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2016-09-14T09:05:59Z

---

This is a SDL2 related issue, which changed significantly the way console can be called on international keyboards.

Potential patch "here":https://github.com/JACoders/OpenJK/commit/0a65fd6a7237f66369e2b3a339b633e0e5f42, with some caveats:

* New cvar: cl_consoleUseScanCode (defaults to 1)
* When this is on it attempts to use whatever the key left of the 1 on all keyboard layouts known by SDL.
* Any language that has a caret symbol (^) as that key will require the shift key again. (AFAIK this is all German layouts)

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2017-02-02T22:56:29Z

---

Someone with a German keyboard (that have a few issues) needs to evaluate the usefulness of this feature.
Otherwise this ticket is going to be closed as wontfix (at least from my side).

**Comment by ETLegacyTeam** [comment]
From: @Kissaki (Kissaki)
Date: 2017-02-03T15:24:59Z

---

Evaluate the usefulness of what feature? This ticket describes an issue, not a (new) feature.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2017-03-04T23:16:21Z

---

Vanilla ET hardcodes several keyboards for making this console key working as one would expect (german, french, spanish..). Of course, any other keyboard would fail. SDL gives a key that works on all keyboards whatever the keymap used, so this is an improvement.

The proposed patch above doesn't work for German keyboard (because ^ is located on the same top left key required for console), so this is not a fix for German people (more of a feature for other people that do not want to change habits). If anything, I'd consider the current SDL solution as the real fix as we won't hardcode keymap.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2017-10-19T08:43:47Z

---

From Keymon:

Btw. I know I also was strongly for getting back the possibility to open the console with [^] on German keyboard, but I just wanted to say that I take that back. Adding hacks into the code just to compensate old habits is a bit over-the-top. I even noticed that I adjusted my habits already and now I try to open the console in vanilla also with [SHIFT]+[ESC]. Habits are a weird thing...
https://dev.etlegacy.com/issues/937
From my perspective this can be set to invalid.

This is also my opinion. German keyboard users can adapt with a little bit of time, no ugly half working workaround needed. Closing.
